source/extensions/filters/http/composite/action.cc

```diff
@@ -9,6 +9,73 @@ void ExecuteFilterAction::createFilters(Http::FilterChainFactoryCallbacks& callb
   cb_(callbacks);
 }
 
+float ExecuteFilterActionFactory::getSampleRatio(
+    const envoy::extensions::filters::http::composite::v3::ExecuteFilterAction& composite_action) {
+  // In default case, if sample_percent is not populated, sample_ratio is 100%, i.e, all sampled.
+  if (!composite_action.has_sample_percent()) {
+    // 1, i.e, 100%,  means all sampled.
+    return 1;
+  }
+  const auto& sample_percent = composite_action.sample_percent().default_value();
+  float denominator = 100;
+  switch (sample_percent.denominator()) {
+  case envoy::type::v3::FractionalPercent::HUNDRED:
+    denominator = 100;
+    break;
+  case envoy::type::v3::FractionalPercent::TEN_THOUSAND:
+    denominator = 10000;
+    break;
+  case envoy::type::v3::FractionalPercent::MILLION:
+    denominator = 1000000;
+    break;
+  default:
+    throw EnvoyException(
+        fmt::format("ExecuteFilterAction sample_percent config denominator setting "
+                    "is invalid : {}. Valid range 0~2.",
+                    sample_percent.denominator()));
+  }
+
+  float sample_ratio = static_cast<float>(sample_percent.numerator()) / denominator;
+  if (sample_ratio > 1) {
+    throw EnvoyException(fmt::format(
+        "ExecuteFilterAction sample_percent config is invalid. sample_ratio={}(Numerator {} / "
+        "Denominator {}). The valid range is 0~1.",
+        sample_ratio, sample_percent.numerator(), denominator));
+  }
```

Comment:
I'm very surprised this type of error checking doesn't already exist elsewhere in the code base. If it doesn't can you move it to where the other runtime helpers live?

Reply:
Done